demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,508 @@
+                THROW_IE_EXCEPTION
+                        << "Every device used with the imagenet_classification_demo should "
+                        << "support OPTIMAL_NUMBER_OF_INFER_REQUESTS ExecutableNetwork metric. "
+                        << "Failed to query the metric for the " << FLAGS_d << " with error:" << ex.what();
+            }
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // ---------------------------------------Create infer request----------------------------------------
+        std::vector<InferRequest> inferRequests;
+        int infReqNum = FLAGS_nireq;
+        for (int infReqID = 0; infReqID < infReqNum; ++infReqID) {
+            inferRequests.push_back(executableNetwork.CreateInferRequest());
+        }
+        // ---------------------------------------------------------------------------------------------------
+        
+        // ----------------------------------------Create output info-----------------------------------------
+        Presenter presenter(FLAGS_u, 0);
+        int width;
+        int height;
+        std::vector<std::string> gridMatRowsCols = split(FLAGS_res, 'x');        
+        if (gridMatRowsCols.size() != 2) {
+            throw std::runtime_error("The value of GridMat resolution flag is not valid.");
+        } else {
+            width = std::stoi(gridMatRowsCols[0]);
+            height = std::stoi(gridMatRowsCols[1]);
+        }
+        GridMat gridMat = GridMat(inputImagesCount, presenter, cv::Size(width, height));
+
+        // image, predicted label, isPredictionRight
+        std::list<std::tuple<cv::Mat, std::string, bool>> shownImagesInfo;
+        // ---------------------------------------------------------------------------------------------------
+        
+        // -----------------------------Prepare variables and data for main loop------------------------------
+        double avgFPS = 0;
+        double avgLatency = 0;
+        double latencySum = 0;
+        unsigned framesNum = 0;
+        long long rightPredictionsCount = 0;
+        long long totalPredictionsCount = 0;
+        double accuracy;
+        bool isTestMode = true;
+        int delay = 1;
+        char key = 0;
+        std::size_t nextImageIndex = 0;
+        long long startTickCount = cv::getTickCount();
+        std::condition_variable condVar;
+        std::mutex mutex;
+        std::exception_ptr irCallbackException;
+        
+        std::queue<IRInfo> emptyInferRequests;
+        std::queue<IRInfo> completedInferRequests;
+        for (std::size_t i = 0; i < inferRequests.size(); i++) {
+            emptyInferRequests.push(IRInfo(inferRequests[i], std::vector<IRInfo::IRImage>()));
+        }
+
+        auto startTime = std::chrono::system_clock::now();
+        auto currentTime = std::chrono::system_clock::now();
+        std::chrono::duration<double> elapsedSeconds = currentTime - startTime;
```

Comment:
```suggestion
        std::chrono::steady_clock::duration elapsedSeconds = currentTime - startTime;
```
I don't see why it should be double.